src/evaluators/BinaryExpression.js

```diff
@@ -170,8 +177,14 @@ export function computeBinary(
 
   if (lval instanceof AbstractValue || rval instanceof AbstractValue) {
     let type = getPureBinaryOperationResultType(realm, op, lval, rval, lloc, rloc);
-    return realm.createAbstract(new TypesDomain(type), ValuesDomain.topVal, [lval, rval], ([lnode, rnode]) =>
-      t.binaryExpression(op, lnode, rnode)
+    let [hash, args] = hashBinary(op, lval, rval);
+    return realm.createAbstract(
+      new TypesDomain(type),
+      ValuesDomain.topVal,
+      hash,
+      args,
+      ([lnode, rnode]) => t.binaryExpression(op, lnode, rnode),
+      op
```

Comment:
Doing a structural equals between different AbstractValue instances can only be done safely by taking the operator that resulted in the abstract value into account along with the arguments. The operator is usually part of the buildNode, but it is easier and probably safer to get to it via the kind property.